MetaMask/metamask-extension
# feat: Confirmation Redesign "Confirm" button should be red when transaction is malicious

## **Description**

The confirm button should be red when the transaction is identified as malicious. There is ongoing work (Alerts) that will replace logic from this PR in the near future. This is the reason why full test coverage was not added here. 

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2448

## **Manual testing steps**

1. set `ENABLE_CONFIRMATION_REDESIGN=true` in .metamaskrc
2. enable Experimental > Redesign Confirmation in settings
3. test malicious signature e.g. PPOM - Malicious Permit in test-dapp
5. observe red button

## **Screenshots/Recordings**



### **Before**



### **After**

<img height="640" alt="Screenshot 2024-05-07 at 6 31 10 AM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/bd3938f0-032a-4ba1-978e-0b4525fb51cb">
<img height="640" alt="Screenshot 2024-05-07 at 6 19 33 AM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/93bf3dad-a4b1-4b3f-87d1-4947582fcd80">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
